### DISCUSSION TITLE
Is it forbidden to use @SdkProtectedApi outside of the SDK?

### DISCUSSION THREAD

**Discussion by mtheos** [aws/aws-sdk-java-v2#5296] [Q&A] [answered]
Hi team,

Is it forbidden to use `@SdkProtectedApi` classes outside of the SDK?

The SDK has a `ResettableInputStream` class for wrapping `FileInputStreams` that allows files to be reset for retries.

We want to use it with the SDK, but we want to expose the class in our s3 facade so we can maintain an `InputStream` interface on our facade.

Thanks

**Comment by sugmanue** [comment, marked as answer]
It's not *forbidden*, but we don't make any backwards compatibility promises as we would do for `@SdkPublicApi` and reserve the right to change it in non-backwards compatible ways or even remove it if not longer needed. So, you can use it, but at your own peril.
